pkg/kubectl/swagger_doc.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubectl
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	// "github.com/davecgh/go-spew/spew"
+	"github.com/emicklei/go-restful/swagger"
+
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+// GetModelDescription returns the description of a specific model or dot path
+func GetModelDescription(inModel, apiVer string, w io.Writer, kubeClient client.Interface) {
+	swaggerSchema, err := kubeClient.SwaggerSchema(apiVer)
+	if err != nil {
+		fmt.Printf("Couldn't read swagger schema from server: %v\n", err)
+		os.Exit(1)
+	}
+	apiVer = swaggerSchema.ApiVersion + "."
+
+	inModel, fieldsPath := splitDotNotation(inModel)
+
+	var pointedModel *swagger.NamedModel
+	allModels := make(map[string]*swagger.NamedModel)
+	for i := range swaggerSchema.Models.List {
+		name := swaggerSchema.Models.List[i].Name
+
+		allModels[name] = &swaggerSchema.Models.List[i]
+		if name == apiVer+inModel {
+			pointedModel = &swaggerSchema.Models.List[i]
+		}
+	}
+	if pointedModel == nil {
+		fmt.Fprintf(w, "No model found with name: %s\n", inModel)
+		return
+	}
+
+	for _, field := range fieldsPath {
+		if prop := findField(pointedModel, field); prop != nil {
+			next := getNextModel(prop)
+			if next == "" {
+				fmt.Fprintf(w, "Field: %s can not be expanded further.\n", field)
+				return
+			}
+			pointedModel = allModels[next]
+		} else {
+			fmt.Fprintf(w, "Field: %s doesn't exist.\n", field)
+			return
+		}
+	}
+	printModelInfo(w, pointedModel, apiVer)
+}
+
+func splitDotNotation(model string) (string, []string) {
+	var fieldsPath []string
+	dotModel := strings.Split(model, ".")
+	if len(dotModel) >= 1 {
+		fieldsPath = dotModel[1:]
+	}
+	return dotModel[0], fieldsPath
+}
+
+func getNextModel(prop *swagger.ModelProperty) string {
+	if prop.Ref != nil {
+		return *prop.Ref
+	} else if *prop.Type == "array" && prop.Items.Ref != nil {
+		return *prop.Items.Ref
+	}
+	return ""
```

Comment:
This will give an error when inModel is a primitive type field, for ex: Pod.PodSpec.RestartPolicy.
Is that intended? We could just return the swagger description for that field.